packages/syft/src/syft/service/settings/settings_service.py

```diff
@@ -111,42 +95,50 @@ def _update(
         SyftSuccess: Settings updated successfully.
         """
+
         result = self.stash.get_all(context.credentials)
-        if result.is_ok():
-            current_settings = result.ok()
-            if len(current_settings) > 0:
-                new_settings = current_settings[0].model_copy(
-                    update=settings.to_dict(exclude_empty=True)
-                )
-                notifier_service = context.server.get_service("notifierservice")
+        if result.is_err():
+            return SyftError(message=result.err())
 
-                # If notifications_enabled is present in the update, we need to update the notifier settings
-                if settings.notifications_enabled is True:
-                    if not notifier_service.settings(context):
-                        return SyftError(
-                            message="Create notification settings using enable_notifications from user_service"
-                        )
-                    notifier_service = context.server.get_service("notifierservice")
-                    result = notifier_service.set_notifier_active_to_true(context)
-                elif settings.notifications_enabled is False:
-                    if not notifier_service.settings(context):
-                        return SyftError(
-                            message="Create notification settings using enable_notifications from user_service"
-                        )
-                    notifier_service = context.server.get_service("notifierservice")
-                    result = notifier_service.set_notifier_active_to_false(context)
-                update_result = self.stash.update(context.credentials, new_settings)
-                return update_result
-            else:
-                return Err(value="No settings found")
+        current_settings = result.ok()
+        if len(current_settings) == 0:
+            return SyftError(message="No settings found")
+        try:
+            new_settings = current_settings[0].model_copy(
+                update=settings.to_dict(exclude_empty=True)
+            )
+            ServerSettings.model_validate(new_settings.to_dict())
+        except ValidationError as e:
+            return SyftError(message=str(e))
+
+        notifier_service = context.server.get_service("notifierservice")
+        if not notifier_service.settings(context):
+            return SyftError(
+                message="Create notification settings using enable_notifications from user_service"
+            )
+        # If notifications_enabled is present in the update, we need to update the notifier settings
+        if settings.notifications_enabled is True:
+            result = notifier_service.set_notifier_active_to_true(context)
+        elif settings.notifications_enabled is False:
+            result = notifier_service.set_notifier_active_to_false(context)
```

Comment:
This is a little bit unrelated to this PR, but we can replace `set_notifier_active_to_false` and `set_notifier_active_to_true` with just 1 function `set_notifier_active(yes: bool)` and then this could be just `result = notifier_service.set_notifier_active(settings.notifications_enabled)`.